pkg/services/sqlstore/user.go

```diff
@@ -132,6 +132,37 @@ func (ss *SQLStore) createUser(ctx context.Context, sess *DBSession, args user.C
 	return usr, nil
 }
 
+func (ss *SQLStore) resetAdminUser(ctx context.Context, sess *DBSession, args user.CreateUserCommand) (user.User, error) {
+	usr := user.User{
+		Login:   args.Login,
+		Email:   args.Email,
+		IsAdmin: true,
+	}
+
+	salt, err := util.GetRandomString(10)
+	if err != nil {
+		return usr, err
+	}
+	usr.Salt = salt
+	rands, err := util.GetRandomString(10)
+	if err != nil {
+		return usr, err
+	}
+	usr.Rands = rands
+
+	if len(args.Password) > 0 {
+		encodedPassword, err := util.EncodePassword(args.Password, usr.Salt)
+		if err != nil {
+			return usr, err
+		}
+		usr.Password = encodedPassword
+	}
+
+	_, err = sess.ID(1).Update(usr)
```

Comment:
It's not safe to assume that the admin user ID is always 1 - it most likely is, especially in the use cases you had in mind when adding this configuration, but it could be used in an instance where the admin user ID is not 1. We will need to verify that user is the admin. 

You can compare this to the reset-admin-password CLI command, which also assumes `1` is the default admin user ID, but validates that by first getting the user and checking the is_admin field: https://github.com/grafana/grafana/blob/main/pkg/cmd/grafana-cli/commands/reset_password_command.go#L53

Also, take a look at the user service [UpdatePassword](https://github.com/grafana/grafana/blob/main/pkg/services/user/userimpl/store.go#L366) method; your Update command should look closer to that one.

----

Please let me know if you need more specific suggestions, or even if you want me to push a commit to your PR branch - I don't want to over-explain and I'm not in a rush, but I also don't want you to feel abandoned (again) :)

Reply:
It would be really nice if you could push commit to the branch.

I just hope we can have this feature soon

Reply:
Sure thing! Just make sure you've [Allowed edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) on your fork and I'll see what I can do 😁